### ISSUE TITLE
TreeSelect nzCheckable and nzCheckStrictly causes parent node to be removed when only child node is clicked

### ISSUE THREAD

**Issue by atom-morgan** [NG-ZORRO/ng-zorro-antd#5068]
### Reproduction link
[https://stackblitz.com/edit/angular-ogz5wh?file=src/app/app.component.ts](https://stackblitz.com/edit/angular-ogz5wh?file=src/app/app.component.ts)

### Steps to reproduce
1. Click nodes `parent 1-0`, `leaf 1-0-0`, and `leaf 1-0-1`.
1. Click to remove node `leaf 1-0-0`.

Only node `leaf 1-0-0` is removed from the list.

Update the template code to what's below which adds `nzCheckable`.

```
<nz-tree-select
  style="width: 250px"
  nzPlaceHolder="Please select"
  [(ngModel)]="value"
  [nzMaxTagCount]="3"
  [nzMaxTagPlaceholder]="omittedPlaceHolder"
  [nzNodes]="nodes"
  [nzDefaultExpandAll]="true"
  [nzAllowClear]="false"
  [nzMultiple]="true"
  [nzCheckStrictly]="true"
  [nzCheckable]="true"
  (ngModelChange)="onChange($event)"
  >
</nz-tree-select>
```

1. Click nodes `parent 1-0`, `leaf 1-0-0`, and `leaf 1-0-1`.
1. Click to remove node `leaf 1-0-0`.

Nodes `parent 1-0` and `leaf 1-0-0` are both removed from the list.

### What is expected?
Only `leaf 1-0-0` should be removed from the list

### What is actually happening?
Nodes `parent 1-0` and `leaf 1-0-0` are both removed from the list.

| Environment | Info |
|---|---|
| ng-zorro-antd | 9.0.0 |
| Browser | Chrome 80.0.3987.163 |

---
If I understand the documentation correctly, `nzCheckable` should simply add checkboxes to the `TreeSelect` and `nzCheckStrictly` should remove any associations between parent nodes and child nodes but using these two together doesn't seem to work correctly.

**Comment by vthinkxie** [comment]
Hi @atom-morgan 
I have tried to replace the template as you described, but everything works fine
https://stackblitz.com/edit/angular-ogz5wh-qipucw?file=src/app/app.component.ts

**Comment by atom-morgan** [comment]
Hi @vthinkxie, I just tried this again and I'm still getting the same results. I've included a gif of the behavior that I'm seeing.

![ant-tree](https://user-images.githubusercontent.com/3698634/79867932-23b70800-83ad-11ea-97f4-4c2c35da7f85.gif)

**Comment by simplejason** [comment]
It only happens under `nzCheckable` mode, we will fix it in #5389
